Public/Src/Sandbox/MacOs/Sandbox/Src/SandboxedPip.hpp

```diff
@@ -157,21 +173,38 @@ class SandboxedPip : public OSObject
      * If no such record exists, a new one is created and associated with the path.
      * Return value of NULL indicates that there is an inherent reason why the path cannot be added to cache.
      */
-    inline CacheRecord* cacheLookup(const char *path)
+    CacheRecord* cacheLookup(const char *path)
     {
-        if (!g_bxl_enable_cache)
+        AutoIncDec callCnt(&cacheCallCnt_);
+
+        // check if we dynamically decided to disable caching for this pip
+        if (RefreshDisableCaching())
         {
+            // If caching is disabled, and no one else is using pathCache_, and we came here first:
+            //   --> safe to release old path cache left to be garbage collected
```

Comment:
```suggestion
            //   --> safe to release old path cache left to be garbage collected
            // Reason: even if someone else comes in after we've checked `cacheCallCnt_ == 1`, 
            //         because once caching is disabled it stays disabled, they'll get to use the new path 
            //         cache object, so it's still safe to release oldPathCache_)
```